StepsToCare/CareEntry.ascx.cs

```diff
@@ -939,6 +1199,162 @@ private void BindAssignedPersonsGrid()
+                var category = DefinedValueCache.Get( categoryId.Value );
+                var categoryFollowUpAfter = category.GetAttributeValue( "FollowUpAfter" ).AsIntegerOrNull();
+                var categoryTimesToRepeat = category.GetAttributeValue( "TimesToRepeat" ).AsIntegerOrNull();
+                if ( categoryFollowUpAfter.HasValue && categoryFollowUpAfter > 0 )
+                {
+                    cbCustomFollowUp.Checked = true;
+                    numbRepeatDays.IntegerValue = categoryFollowUpAfter;
+                    numbRepeatTimes.IntegerValue = categoryTimesToRepeat;
+                    pnlRecurrenceOptions.Visible = true;
+                }
+            }
+            if ( previewAssignedPeople && categoryId.HasValue && person != null && needId == 0 && dateDifference <= futureThresholdDays )
+            {
+                var autoAssignWorker = GetAttributeValue( AttributeKey.AutoAssignWorker ).AsBoolean();
+                var autoAssignWorkerGeofence = GetAttributeValue( AttributeKey.AutoAssignWorkerGeofence ).AsBoolean();
+                var loadBalanceType = GetAttributeValue( AttributeKey.LoadBalanceWorkersType );
+                var enableLogging = GetAttributeValue( AttributeKey.VerboseLogging ).AsBoolean();
+                var leaderRoleGuids = GetAttributeValues( AttributeKey.GroupTypeAndRole ).AsGuidList();
+
+                CareNeed careNeed = GenerateTempNeed( person, categoryId );
+
+                AssignedPersons = CareUtilities.AutoAssignWorkers( careNeed, cbWorkersOnly.Checked, autoAssignWorker: autoAssignWorker, autoAssignWorkerGeofence: autoAssignWorkerGeofence, loadBalanceType: loadBalanceType, enableLogging: enableLogging, leaderRoleGuids: leaderRoleGuids, previewAssigned: previewAssignedPeople );
+                pwAssigned.Visible = UserCanAdministrate;
+                BindAssignedPersonsGrid();
+            } else
+            {
+                pwAssigned.Visible = false;
+                AssignedPersons = null;
+            }
+        }
+
+        private CareNeed GenerateTempNeed( Person person, int? categoryId = null )
+        {
+            if ( categoryId == null )
+            {
+                categoryId = dvpCategory.SelectedValue.AsIntegerOrNull();
+            }
+            var careNeed = new CareNeed { Id = 0 };
+            careNeed.CampusId = cpCampus.SelectedCampusId;
+            careNeed.PersonAlias = person?.PrimaryAlias;
+            careNeed.PersonAliasId = person?.PrimaryAliasId;
+            careNeed.StatusValueId = dvpStatus.SelectedValue.AsIntegerOrNull();
+            careNeed.CategoryValueId = categoryId;
+
+            careNeed.LoadAttributes();
+
+            phAttributes.Controls.Clear();
+            Helper.AddEditControls( careNeed, phAttributes, false, BlockValidationGroup, 2 );
+
+            return careNeed;
+        }
+
+        private bool createNote( RockContext rockContext, int entityId, string noteText, bool countsForTouch = false )
```

Comment:
Can this be in StepsToCare Utilities instead? Seems like a duplicate method from CareDashboard.

Reply:
No, it turns out it uses a lot of built in Rock block variables, such as CurrentPerson, Authorization and some other items. Not worth passing all of it in.